rfcs/bring-ts.md

```diff
@@ -0,0 +1,267 @@
+# Bring TypeScript
+
+- **Author(s):**: @eladb
+- **Submission Date**: {2023-01-03}
+- **Stage**: Draft
+
+**Goal**: allow developers to use any TypeScript library in their Wing applications.
+
+## Feature Spec
+
+Wing has great interoperability with the TypeScript ecosystem. You can bring any TypeScript library
+and use it in Wing.
+
+### How does Wing bring typescript libraries?
+
+When the Wing compiler sees something like this:
+
+```js
+bring colors;
+```
+
+It will use node's standard module resolution algorithm to lookup for `colors`. When found, it will
+check if there's a `.jsii` manifest in the module. Since there isn't, it will check if the directory `vendor/colors.wlib`
+exists.
+
+This directory is expected to include a `.jsii` file and an `index.js` file.
+
+### ts2wing
+
+In order to use a TypeScript library, we will use the `ts2wing` command to create a Wing Library
+(`.wlib`) from it.
+
+Install `ts2wing` on your system (maybe in the future it will be included in the wing toolchain):
+
+```sh
+$ npm i @winglang/ts2wing
+$ ts2wing --version
+0.1.2
+```
+
+Let's say I want to use the [colors](https://www.npmjs.com/package/colors) library in my Wing code.
+
+First, I am going to install it via npm/yarm/pnpm or whatever:
+
+```sh
+npm i colors
+```
+
+Now, I am going to generate bindings for it using `ts2wing`:
+
+```sh
+$ ts2wing colors
+Found colors@1.4.0 under node_modules/colors...
+Reading definitions from index.d.ts...
+Generating Wing wrapper classes under vendor/colors/index.ts...
+Compiling JSII to vendor/colors/.jsii...
+Generating vendor/colors/API.md...
+```
+
+### The .wlib Directory
+
+This command created a Wing Library file under `vendor/colors.wlib`.
+
+> A `.wlib` file it a self-contained artifact that represents a reusable library in Wing. It
+> includes all the dependencies needed by the library (vendored in). (see note below about future
+> support for [public dependencies](#public-dependencies-are-not-supported)).
+
+Let's take a look at the API of our new library:
+
+```sh
+$ wing docs colors
+Opening browser...
+```
+
+This will open your browser with the generated API documentation for the generated library. Cute,
+ha?
+
+### Consuming the library
+
+OK, let's use this library in our Wing code.
+
+```js
+bring colors;
+
+print("${colors.Global.green("hello")}, ${colors.Colors.blue("world")}!")
+```
+
+As you can see, since Wing doesn't support global functions, the `Global` wrapper class was
+generated and includes a static method for each public API exported from `colors`.
+
+Similarly to turning global functions to static methods, `ts2wing` applies other heuristics to
+create an [adaptation layer](#type-system-adapters) between the type systems.
+
+### Escape hatch
+
+As you might have expected, there could be cases where `ts2wing` won't be able to create an adapter
```

Comment:
Will a user be made aware of when the wing compiler can't create the necessary adapters? You could take this further and prompt them to use the edit command